README.md

```diff
@@ -1,36 +1,101 @@
-# Cairo VM in Typescript
+# TypeScript Cairo VM
+
+<div align="center">
+  <h1><code>TypeScript Cairo VM</code></h1>
+
+<strong>An implementation of the Cairo VM in TypeScript, focusing on
+education</strong>
+
+[Github](https://github.com/kkrt-labs/cairo-vm-ts)
+
+<sub>Built with 🥕 by <a href="https://twitter.com/KakarotZkEvm">KKRT Labs</a></sub>
+
+</div>
+
+> ⚠️ This project is in its genesis, undergoing fast-paced development. It is
+> not suitable for production yet. Expect frequent breaking changes.
 
 ## What is Cairo
 
-<!-- TODO: fill this section -->
+Cairo stands for CPU AIR "o" (like insp"o", conv"o", 🤔).
 
-Cairo stands for CPU AIR "o" (like insp"o", conv"o", 🤔). See
-[Cairo whitepaper](https://eprint.iacr.org/2021/1063) for more information.
+It is a framework (a machine, an assembly and a language) which allows writing provable programs without having to understand the underneath ZK-technology.
+
+A program written in Cairo (or Cairo Zero) is compiled and then executed on a Cairo VM which produces traces to be used for the STARK proof generation.
+
+See [resources](#resources) for more information.
 
 ## Why this
 
-There are currently three Cairo VM implementations:
+### Client Diversity
```

Comment:
I'm not sure that I would put client diversity, this term being rather reserved for blockchain clients while we are talking here about the virtual machine itself and not the full client. I would say maybe simply implementation diversity or virtual machine diversity, something like that.

Reply:
Indeed, "implementation diversity" is a better term